env0/resource_environment.go

```diff
@@ -0,0 +1,184 @@
+package env0
+
+import (
+	"context"
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceEnvironment() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceEnvironmentCreate,
+		ReadContext:   resourceEnvironmentRead,
+		UpdateContext: resourceEnvironmentUpdate,
+		DeleteContext: resourceEnvironmentDelete,
+
+		Importer: &schema.ResourceImporter{StateContext: resourceTeamImport},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Description: "the environment's id",
+				Optional:    true,
+				Computed:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The environment's name",
+				Required:    true,
+			},
+			"project_id": {
+				Type:        schema.TypeString,
+				Description: "the environment's project id",
+				Required:    true,
+			},
+			"template_id": {
+				Type:        schema.TypeString,
+				Description: "the environment's template id",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func setEnvironmentSchema(d *schema.ResourceData, environment client.Environment) {
+	d.Set("name", environment.Name)
+	d.Set("project_id", environment.ProjectId)
+	d.Set("template_id", environment.TemplateId)
+}
+
+func resourceEnvironmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	payload := getCreatePayload(d)
+
+	environment, err := apiClient.EnvironmentCreate(payload)
+	if err != nil {
+		return diag.Errorf("could not create environment: %v", err)
+	}
+
+	d.SetId(environment.Id)
+	setEnvironmentSchema(d, environment)
+
+	return nil
+}
+
+func resourceEnvironmentRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	environment, err := apiClient.Environment(d.Id())
+	if err != nil {
+		return diag.Errorf("could not get environment: %v", err)
+	}
+
+	setEnvironmentSchema(d, environment)
+
+	return nil
+}
+
+func resourceEnvironmentUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	payload := getUpdatePayload(d)
+
+	// TODO: deploy if needed
+
+	_, err := apiClient.EnvironmentUpdate(d.Id(), payload)
+	if err != nil {
+		return diag.Errorf("could not update environment: %v", err)
+	}
+
+	return nil
+}
+
+func resourceEnvironmentDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	_, err := apiClient.EnvironmentDestroy(d.Id())
+	if err != nil {
+		return diag.Errorf("could not delete team: %v", err)
```

Comment:
copy 🍝 
`could not delete team` -> `could not destroy environment`

Reply:
903ab6d